### ISSUE TITLE
Mismatched types

### ISSUE THREAD

**Issue by tolusophy** [mc2-project/delphi#29]
**I got this error while trying to compile neural networks**

I tried to change into_iter() to iter() [just guessed it could be the solution]... but it didn't work. I have checked all sources I could check online and I have no found a solution yet.

Please, what could be done about it?

**Comment by tolusophy** [comment]
```
Compiling neural-network v0.1.0 (/home/toluwani/delphi/rust/neural-network)
error[E0308]: mismatched types
  --> neural-network/src/tensors/macros.rs:73:17
   |
72 |             fn into_iter(self) -> Self::IntoIter {
   |                                   -------------- expected `ndarray::iter::Iter<'_, F, Dim<[usize; 4]>>` because of return type
73 |                 self.0.into_iter()
   |                 ^^^^^^^^^^^^^^^^^^ expected struct `ndarray::iter::Iter`, found struct `ndarray::iterators::into_iter::IntoIter`
   |
  ::: neural-network/src/tensors/mod.rs:14:1
   |
14 | ndarray_impl!(Input, Array4, Quadruple);
   | --------------------------------------- in this macro invocation
   |
   = note: expected struct `ndarray::iter::Iter<'_, F, Dim<[usize; 4]>>`
              found struct `ndarray::iterators::into_iter::IntoIter<F, Dim<[usize; 4]>>`
   = note: this error originates in the macro `ndarray_impl` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> neural-network/src/tensors/macros.rs:73:17
   |
72 |             fn into_iter(self) -> Self::IntoIter {
   |                                   -------------- expected `ndarray::iter::Iter<'_, F, Dim<[usize; 4]>>` because of return type
73 |                 self.0.into_iter()
   |                 ^^^^^^^^^^^^^^^^^^ expected struct `ndarray::iter::Iter`, found struct `ndarray::iterators::into_iter::IntoIter`
   |
  ::: neural-network/src/tensors/mod.rs:15:1
   |
15 | ndarray_impl!(Kernel, Array4, Quadruple);
   | ---------------------------------------- in this macro invocation
   |
   = note: expected struct `ndarray::iter::Iter<'_, F, Dim<[usize; 4]>>`
              found struct `ndarray::iterators::into_iter::IntoIter<F, Dim<[usize; 4]>>`
   = note: this error originates in the macro `ndarray_impl` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0308`.
error: could not compile `neural-network` due to 2 previous errors
```

**Comment by Ti-Oluwanimi** [comment]
I tried to change into_iter() to iter() [just guessed it could be the solution]... but it didn't work.

What could be done about it?

**Comment by Ti-Oluwanimi** [comment]
Replies? any?

**Comment by ryanleh** [comment]
Hi,
Could you see if replacing the line in question with:
```
(&self.0).into_iter()
```
works for you?

**Comment by tolusophy** [comment]
@ryanleh Thank you so much. Yes, no more issues with the neural_network, but I think it triggered slice in protocol_sys

```
Compiling protocols-sys v0.1.0 (/home/toluwani/delphi/rust/protocols-sys)
error[E0599]: no method named `into_slice` found for struct `ArrayBase<ViewRepr<&u64>, Dim<[usize; 2]>>` in the current scope
  --> protocols-sys/src/server_cg.rs:90:22
   |
90 |                     .into_slice()
   |                      ^^^^^^^^^^ help: there is an associated function with a similar name: `slice`

error[E0599]: no method named `into_slice` found for struct `ArrayBase<ViewRepr<&u64>, Dim<[usize; 3]>>` in the current scope
   --> protocols-sys/src/server_cg.rs:165:18
    |
165 |                 .into_slice()
    |                  ^^^^^^^^^^ help: there is an associated function with a similar name: `slice`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `protocols-sys` due to 2 previous errors
```

I changed it to slice() and it brought out more errors, so I had to change it back to ```into_slice```.  Any ideas while I also try to determine what could be wrong. 

**Thanks, once again!**

**Comment by ryanleh** [comment]
I just pushed all the necessary fixes for updating `ndarray` to `master` in #30 -- please re-pull and try building again.

**Comment by tolusophy** [comment]
Alright! Thank you @ryanleh
